### PR TITLE
Add Wild Apricot integration attempt 2

### PR DESCRIPTION
**Pull request by MutatedGamer**

### CODE REVIEW

File: api/src/API/Parameters/OrgAssignedMemberIdAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+using System;
+
+namespace RaceResults.Api.Parameters
+{
+    [AttributeUsage(AttributeTargets.Parameter)]
```
Comment (MutatedGamer) [inline api/src/API/Parameters/OrgAssignedMemberIdAttribute.cs:5]: This can be removed

File: api/src/Common/Models/Organization.cs [hunk 2]
```diff
@@ -10,6 +10,15 @@ public struct Organization : IModel
         [Required]
         public string Name { get; set; }
 
+        [Required]
+        public string WildApricotDomain { get; set; }
```
Comment (MutatedGamer) [inline api/src/Common/Models/Organization.cs:14]: Remove these two

File: api/src/Common/Models/RaceResult.cs [hunk 3]
```diff
@@ -1,5 +1,6 @@
 using System;
 using System.ComponentModel.DataAnnotations;
+using System.Text.Json.Serialization;
```
Comment (MutatedGamer) [inline api/src/Common/Models/RaceResult.cs:3]: Remove

**Review by luillo1 — DISMISSED** [review]

File: api/src/API/Controllers/MembersController.cs [hunk 4]
```diff
@@ -28,38 +31,64 @@ public class MembersController : ControllerBase
...
             return Ok(result);
         }
 
-        [HttpGet("{memberId}")]
-        public async Task<IActionResult> GetOneMember(string orgId, string memberId)
-        {
-            MemberContainerClient container = containerProvider.MemberContainer;
-            Member result = await container.GetOneAsync(memberId, orgId);
-            return Ok(result);
-        }
```
[the remark this answers is not in the document]
Comment (MutatedGamer) [reply api/src/API/Controllers/MembersController.cs:44]: Don't think so

File: api/src/API/Parameters/OrganizationIdAttribute.cs [hunk 5]
```diff
@@ -0,0 +1,9 @@
+using System;
+
+namespace RaceResults.Api.Parameters
+{
+    [AttributeUsage(AttributeTargets.Parameter)]
+    public class OrganizationIdAttribute : Attribute
```
[the remark this answers is not in the document]
Comment (MutatedGamer) [reply api/src/API/Parameters/OrganizationIdAttribute.cs:6]: This is so we can identify the parameter value in the RequireOrganizationAuth attribute. I'll add a comment here

File: api/src/API/MemberProviders/WildApricot/WildApricotOauthResponse.cs [hunk 6]
```diff
@@ -0,0 +1,24 @@
+using System.Collections.Generic;
+
+namespace RaceResults.Api.MemberProviders.WildApricot
+{
+    public struct WildApricotOauthResponse
+    {
+        public struct WildApricotOauthPermissionsObject
+        {
+            public int accountId { get; set; }
```
[the remark this answers is not in the document]
Comment (MutatedGamer) [reply api/src/API/MemberProviders/WildApricot/WildApricotOauthResponse.cs:9]: i need to check they resharp serializes correctly if I change this

[the remark this answers is not in the document]
Comment (MutatedGamer) [reply api/src/API/MemberProviders/WildApricot/WildApricotOauthResponse.cs:9]: It worked! Huzzah!

File: api/src/API/Controllers/MembersController.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (MutatedGamer) [reply api/src/API/Controllers/MembersController.cs:44]: oh, it's just moved below in the file. it still exists

File: api/src/API/Authorization/RequireOrganizationAuthenticationAttribute.cs [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+    ///     <br />
+    ///     This does not ensure the member is a particular person within that organization.
+    ///     For that, use <see cref="RequireOrganizationAuthorizationAttribute"/>.
+    ///  </summary>
+    public class RequireOrganizationAuthenticationAttribute : ActionFilterAttribute
+    {
+        public const string WildApricotAccountIdHeader = "WA-AccountId";
+        public const string WildApricotAuthorizationHeader = "WA-Authorization";
+
+        protected readonly ICosmosDbContainerProvider containerProvider;
```
Comment (luillo1) [inline api/src/API/Authorization/RequireOrganizationAuthenticationAttribute.cs:27]: stylecop

**Review by luillo1 — APPROVED** [review]

**Pull request merged**